perf: migrate to HEX() uuids

jniles opened the pull request:
BREAKING CHANGE: you should no longer call `uuid()` directly on the server.  Instead, call `util.uuid()` to generate a compatible UUID.

This commit migrates the application to use hexadecimal UUIDs instead of using the standard representation with dashes.  From a data structure perspective, this doesn't change anything - the data is still stored in binary.  However, `SELECT BUID()` is 3x faster due to a simpler construction.

This change caused me to revisit our tests and try to clean up a lot of the code.  Most files were changed to update the syntax for UUIDs as well as clean up lint issues.

Closes #3090.

jniles committed:
perf: migrate to HEX() uuids

BREAKING CHANGE: you should no longer call uuid() directly.  Instead,
call util.uuid() to generate a compatible UUID.

This commit migrates the application to use hexadecimal UUIDs instead of
using the standard representation with dashes.  From a data structure
perspective, this doesn't change anything - the data is still stored in
binary.  However, `SELECT BUID()` is 3x faster due to a simpler
construction.

This change caused me to revisit our tests and try to clean up a lot of
the code.  Most files were changed to update the syntax for UUIDs as
well as clean up lint issues.

Closes #3090.

sfount commented on server/config/routes.js:
```diff
@@ -323,7 +301,7 @@ exports.configure = function configure(app) {
   app.get('/depots/:depotId/distributions', depots.listDistributions);
   app.get('/depots/:depotId/distributions/:uuid', depots.detailDistributions);
-  app.post('/depots/:depotId/distributions', depots.createDistributions);
+  // app.post('/depots/:depotId/distributions', depots.createDistributions);
```
It looks like the `createDistributions` method has been removed from depots, this can probably be fully removed.

sfount commented on server/models/functions.sql:
```diff
@@ -22,11 +22,9 @@ $$
 */
 CREATE FUNCTION BUID(b BINARY(16))
-RETURNS CHAR(36) DETERMINISTIC
+RETURNS CHAR(32) DETERMINISTIC
```
Is it necessary to write a migrate script for the API/ change in function? I'm under the impression the backup system doesn't include functions or stored procedures so it may not be needed, just making sure there's a plan for applying this PR to existing systems.

sfount reviewed (approved):
This is a great refactor, `eslint` thanks you! 

Insightful analysis with https://github.com/IMA-WorldHealth/bhima-2.X/issues/3090 and well executed. 

Just one question about the roll-out plan but the code look good to me.

jniles committed:
chore(db): add BUID migration script

This commit adds the BUID() migration script to remove the previous
version and update the database with a new one.
